From d45fd85ed25d113d082023294b35cc77b9051b7b Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Thu, 27 Aug 2015 00:16:09 -0400
Subject: [PATCH] Rcon Thread Sleep

Make the RCON thread not pound CPU

Fix decompile issue with RCON
---
 src/main/java/net/minecraft/server/RemoteControlSession.java | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/src/main/java/net/minecraft/server/RemoteControlSession.java b/src/main/java/net/minecraft/server/RemoteControlSession.java
index 1b45f81..50197d4 100644
--- a/src/main/java/net/minecraft/server/RemoteControlSession.java
+++ b/src/main/java/net/minecraft/server/RemoteControlSession.java
@@ -90,7 +90,7 @@ public class RemoteControlSession extends RemoteConnectionThread {
                         this.a(l, String.format("Unknown request %s", new Object[] { Integer.toHexString(i1)}));
                         continue;
                     }
-                }
+                } else {try {Thread.sleep(20);} catch (InterruptedException e1) {e1.printStackTrace();}} // EMC
             } catch (SocketTimeoutException sockettimeoutexception) {
                 break;
             } catch (IOException ioexception) {
@@ -98,13 +98,11 @@ public class RemoteControlSession extends RemoteConnectionThread {
             } catch (Exception exception1) {
                 RemoteControlSession.h.error("Exception whilst parsing RCON input", exception1);
                 break;
-            } finally {
-                this.g();
             }
-
-            return;
         }
 
+        this.g(); // EMC
+
     }
 
     private void a(int i, int j, String s) throws IOException {
-- 
1.9.1

